src/Symfony/Component/Process/InputStream.php

```diff
@@ -20,12 +20,15 @@
  */
 class InputStream implements \IteratorAggregate
 {
+    /** @var null|callable */
     private $onEmpty = null;
     private $input = array();
     private $open = true;
 
     /**
      * Sets a callback that is called when the write buffer becomes empty.
+     *
+     * @param callable $onEmpty
```

Comment:
typehinting here is wrong, it shall be `null|callable`

Reply:
`public function onEmpty(callable $onEmpty = null)`
Is not a constructor.
It is public function which can be called with null, whatever you put in phpdoc. It is nullable, simply code was originally created under php5, and this is php5 way of saying that argument is not nullable.
if you want to create doc for input, you need to respect that.

if you think that null shall be forbidden, you shall change method definition, which would be a bc breaker probably.

Reply:
Ups. Sorry was mistaken there :)
Yep you are right - will add `null` as valid doctype as well.